### PR TITLE
GEOPY-621: Run batch 2D inversion using Sweeper class

### PR DESCRIPTION
**Pull request by benk-mira**
**GEOPY-621 - Run batch 2D inversion using Sweeper class**

### CODE REVIEW

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py
Comment (domfournier) [inline geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:111]: Would be great if we could still call this inversion with the `inversion.driver` instead, like the others. Feels like those methods should be part of the Sweeper class, which could append its methods to the `InversionDriver`.

Comment (domfournier) [reply geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:111]: And looks like GEOPY-620 needs come in first, because `sweep` doesn't exist publicly yet.

Comment (benk-mira) [reply geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:111]: yes, let's defer to next sprint

Comment (benk-mira) [reply geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py]: Removed the pseudo3d driver

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/constants.py
Comment (domfournier) [inline geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/constants.py:237]: Hey would you mind changing this here and for the 3D case, zero is bad for conductivity

**Review by domfournier — APPROVED** [review]
Minor change for start value

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/constants.py
Comment (benk-mira) [reply geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/constants.py:237]: the real value (1e-1) from the defaults dictionary overrides this values anyway, but I've replaced everywhere in the dc packages just for show.

File: geoapps/inversion/constants.py
Comment (domfournier) [inline geoapps/inversion/constants.py:157]: Should we not remove those meshing parameters altogether?

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py
Comment (domfournier) [inline geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:55]: For now, until we refactor, we should probablt assigne to `self.data` and `self.window`  to reflect what's happening in the BaseInversionDriver?

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py
Comment (domfournier) [inline geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:71]: Just for clarity, can we just check for the non-pending case and continue instead. Will avoid the super long indentation.

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py
Comment (domfournier) [inline geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:124]: Why do we need to copy over all the originals? Feels like topo is enough here.

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py
Comment (domfournier) [inline geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:136]: ifile is already 2D, don't think we need any of the text replacements.

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py
Comment (domfournier) [inline geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:150]: I don't think this does anything since you just pop and re-insert. Consider the simpler logic suggested above.

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py
Comment (domfournier) [inline geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:153]: Why?

File: geoapps/inversion/line_sweep/driver.py
Comment (domfournier) [inline geoapps/inversion/line_sweep/driver.py:81]: Related to the why? above, technically the driver should already have its group.

File: geoapps/inversion/constants.py
Comment (benk-mira) [reply geoapps/inversion/constants.py:157]: yeah... maybe the octree ones, but we need the ones that are used in the pseudo3d ui.json

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py
Comment (benk-mira) [reply geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:55]: done

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py
Comment (benk-mira) [reply geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:71]: done

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py
Comment (benk-mira) [reply geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:124]: We also need the data object, but I can do the two call separately from the loop. Done

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py
Comment (benk-mira) [reply geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:136]: I ended up with the title and inversion_type of the batch 2d from the self.pseudo_params update loop.  I've bypassed those to params within the loop rather than correcting after

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py
Comment (benk-mira) [reply geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:150]: done

File: geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py
Comment (benk-mira) [reply geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py:153]: Not needed anymore,  cleaned up the .constants file

File: geoapps/inversion/line_sweep/driver.py
Comment (benk-mira) [reply geoapps/inversion/line_sweep/driver.py:81]: done

**Review by domfournier — APPROVED** [review]
Looking good. Let's merge geoh5py and sweeper 621, then re-point the yml files before merging.

**Review by domfournier — APPROVED** [review]
DismalColorfulBeardedcollie-size_restricted

**Pull request merged**
